receiver/k8sobjectsreceiver/README.md

```diff
@@ -36,9 +36,31 @@ The following is example configuration
         mode: watch
         group: events.k8s.io
         namespaces: [default]
+
+  k8sobjects/leader-election:
+    leader_election:
+      enabled: true
+      lock_name: "k8sobjects-leader-election"
+    auth_type: serviceAccount
+    objects:
+      - name: pods
+        mode: pull
+        label_selector: environment in (production),tier in (frontend)
+        field_selector: status.phase=Running
+        interval: 15m
+      - name: events
+        mode: watch
+        group: events.k8s.io
+        namespaces: [default]
 ```
 
 Brief description of configuration properties:
+- `leader_election`: running in Kubernetes with `leader election mode`, this means that multiple instances are running, but only one is active at a time, and if it fails, another one is elected as leader and takes its place.
```

Comment:
I think we should mention the additional access needed by the collector's service account to perform leader election here, and also update the helm chart.

I believe this snippet should cover rbac for leaderelection.
```
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
  - get
  - update
  ```
  
  In case users prefer to make this stricter and are providing their own election ids, they can update the role to something like
```

- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
- apiGroups:
  - coordination.k8s.io
  resourceNames:
  - <ELECTION ID>
  resources:
  - leases
  verbs:
  - get
  - update
  ```

Reply:
> and also update the helm chart.

yes, I will update soon in [opentelemetry-helm-charts](https://github.com/open-telemetry/opentelemetry-helm-charts) repo after this PR get merged. :-P